.github/dependabot.yml

```
@@ -82,9 +82,33 @@ updates:
       time: "04:20"
       timezone: Europe/Paris
     groups:
-      pythons-dependencies:
+      pytest:
+        patterns:
+          - pytest-*
+          - hypothesis
+      pytools:
+        patterns:
+          - ruff
+          - black
+          - mypy
+          - maturin
+          - editorconfig-checker
+          - setuptools
+      pydocs:
+        patterns:
+          - sphinx*
+      pydeps:
         patterns:
           - "*"
+        exclude-patterns:
+          - sphinx*
+          - pytest-*
+          - hypothesis
+          - ruff
+          - black
+          - mypy
+          - maturin
+          - editorconfig-checker
```

It would be nice to avoid duplication. 
Isn't there a syntax to re-use `pytest`, `pytools`, ... in the `pydeps` exclude-patterns?
If not, maybe if we completely remove the `pydeps` group, the deps that do not match the previous groups will be "grouped" together anyway? :thinking:

---

> Isn't there a syntax to re-use [...] `exclude-patterns`

Yes, YAML syntax support anchor to avoid repetition (https://support.atlassian.com/bitbucket-cloud/docs/yaml-anchors/) sadly the parser used by `dependabot-core` don't support that :cry: dependabot/dependabot-core#1582

> If not, [...] if we completely remove the `pydeps` group, the deps that do not match the previous groups will be "grouped" together anyway?

No, the deps that doesn't match a group pattern will have a dedicated PR for it

---

Then maybe list the `exclude-patterns` in the same order just so it is easier to check if a pattern is missing (btw the `setuptools` pattern is missing?)

```suggestion
      pydeps:
        patterns:
          - "*"
        exclude-patterns:
          - pytest-*
          - hypothesis
          - ruff
          - black
          - mypy
          - maturin
          - editorconfig-checker
          - sphinx*
```

---

Done, that's better indeed